File: programmers/lv1/1-완주하지-못한-선수/solution.py
def solution(participant, completion):
    participant.sort() # 시간복잡도를 줄이기 위해 정렬
    completion.sort()
    for i in range(len(participant)) :
        if i >= len(completion) :
            return participant[i]
        if participant[i] != completion[i] :
            return participant[i]

File: programmers/lv1/1-완주하지-못한-선수/Solution.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> com = new HashMap<>();

        for (int i = 0; i < participant.length; i++)
            com.put(participant[i], com.getOrDefault(participant[i], 0) + 1);
        
        for (String c : completion) {
            com.put(c, com.get(c) - 1);
        }
        
        for (String p : participant) {
            if (com.get(p) != 0) return p;
        }
        return "";
    }
}